### PR TITLE
allow using custom update channels

### PR DESCRIPTION
**Pull request by lrusak**
should be good to go, let me know if the strings make sense.

### CODE REVIEW

**Comment by MilhouseVH** [comment]
Something weird is going on when you cancel the `Automatic Updates`, `Update Channel` and `Available Versions` dialogs. Open these dialogs and then click Cancel (or back - you may need to repeat this a second time).

It looks like the System panel is being refreshed, and half of the time the focus is being "lost", returning to the main menu, ie. click on Update Channel, click cancel, click down and (half of the time) instead of the focus moving to `Available Versions` as expected, instead you move from `System` to `Network`...

File: src/resources/lib/modules/system.py [hunk 1]
```diff
@@ -578,16 +660,42 @@ def do_manual_update(self, listItem=None):
...
-            url = self.UPDATE_DOWNLOAD_URL % ('releases', 'releases.json')
-            update_json = json.loads(self.oe.load_url(url))
-            return update_json
+            if url is None:
+                url = self.UPDATE_DOWNLOAD_URL % ('releases', 'releases.json')
+            if url.split('/')[-1] != 'releases.json':
+                url = url + '/releases.json'
+            data = self.oe.load_url(url)
+            if not data is None:
+                update_json = json.loads(self.oe.load_url(url))
```
Comment (MilhouseVH) [inline src/resources/lib/modules/system.py]: This is loading the url a second time if it worked the first time, doubling the network overhead... shouldn't this be:
```
+                update_json = json.loads(data)
```
?

Comment (lrusak) [inline src/resources/lib/modules/system.py]: Yep that will work better

**Comment by MilhouseVH** [comment]
Thanks, updates are working better.

However there's a problem when downloading:
```
22:31:52 3286.522217 T:1591735200   ERROR: ## LibreELEC Addon ## oe::download_file(http://milhouse.libreelec.tv/builds/master/RPi2/LibreELEC-RPi2.arm-8.0-Milhouse-20160613210617-#0613-g64949dd.tar, /storage/.kodi/temp/LibreELEC-RPi2.arm-8.0-Milhouse-20160613210617-#0613-g64949dd.tar) ## ERROR: (HTTPError())
22:31:52 3286.522949 T:1591735200   ERROR: Traceback (most recent call last):
                                              File "/home/neil/projects/LibreELEC.tv/build.LibreELEC-RPi2.arm-8.0-devel/LibreELEC-settings-168c183/.install_pkg/usr/share/kodi/addons/service.libreelec.settings/oe.py", line 284, in download_file
                                              File "/usr/lib/python2.7/urllib2.py", line 154, in urlopen
                                              File "/usr/lib/python2.7/urllib2.py", line 437, in open
                                              File "/usr/lib/python2.7/urllib2.py", line 550, in http_response
                                              File "/usr/lib/python2.7/urllib2.py", line 475, in error
                                              File "/usr/lib/python2.7/urllib2.py", line 409, in _call_chain
                                              File "/usr/lib/python2.7/urllib2.py", line 558, in http_error_default
                                            HTTPError: HTTP Error 404: Not Found
```
I don't think you're escaping/quoting the url so the hash (#) isn't converted to %23.

There's also no visual indication that the download has failed - it fails silently.

**Comment by MilhouseVH** [comment]
It also looks like it has created the download file in `/storage/.kodi/temp` which makes me wonder if the `temp` folder could slowly become full of partially downloaded .tar files. Maybe a better solution would be to use a single filename for the temp file thus overwriting any existing temp file, then use the correct download filename when copying to `/storage/.update`. This way you'd only have at most one tar file hanging around in the `temp` folder.

**Comment by lrusak** [comment]
@MilhouseVH I updated this, it should now work properly if the user has no internet connection or if they input the wrong url.

**Pull request merged**